tests/validation/tests/v3_api/test_istio.py

```diff
@@ -0,0 +1,266 @@
+    istio_gateway_wl = client.by_id_workload('deployment:' + ns.name
+                                             + ':istio-ingressgateway')
+    assert istio_gateway_wl is not None
+    endpoints = istio_gateway_wl['publicEndpoints'][0]
+    gateway_url = endpoints['addresses'][0] + ':' + str(endpoints['port'])
+    curl_cmd = 'curl -s http://' + gateway_url \
+               + '/productpage | grep -o "<title>.*</title>"'
+    result = run_command(curl_cmd)
+    start = time.time()
+    while result != '<title>Simple Bookstore App</title>\n':
+        if time.time() - start > timeout:
+            raise AssertionError(
+                "Timed out and failed to get bookinfo gateway ready")
+        time.sleep(.5)
+        result = run_command(curl_cmd)
+        assert result == '<title>Simple Bookstore App</title>\n'
+    return result
+
+
+def create_and_test_booinfo_services(p_client, ns, timeout=DEFAULT_TIMEOUT):
+    book_info_file_path = istio_path + '/bookinfo.yaml'
+    execute_kubectl_cmd('apply -f ' + book_info_file_path + ' -n '
+                        + ns.name, False)
+    result = execute_kubectl_cmd('get deployment -n' + ns.name, True)
+
+    for wl in result['items']:
+        wait_for_wl_by_id_to_active(p_client, wl['metadata']['uid'], 60)
+
+    rating_pod = execute_kubectl_cmd('get pod -l app=ratings -n' + ns.name)
+    assert len(rating_pod['items']) == 1
+
+    rating_pod_name = rating_pod['items'][0]['metadata']['name']
+    result = execute_kubectl_cmd(
+        'exec -it -n ' + ns.name + ' ' + rating_pod_name
+        + ' -c ratings -- curl productpage:9080/productpage'
+        + ' | grep -o "<title>.*</title>"', False)
+
+    start = time.time()
+    while result != '<title>Simple Bookstore App</title>\n':
+        if time.time() - start > timeout:
+            raise AssertionError(
+                "Timed out and failed to get bookinfo service ready")
+        time.sleep(.5)
+        result = execute_kubectl_cmd(
+            'exec -it -n ' + ns.name + ' ' + rating_pod_name
+            + ' -c ratings -- curl productpage:9080/productpage'
+            + ' | grep -o "<title>.*</title>"', False)
+        assert result == '<title>Simple Bookstore App</title>\n'
+    return result
+
+
+def wait_for_wl_by_id_to_active(client, workload_uuid,
+                                timeout=DEFAULT_TIMEOUT):
+    start = time.time()
+    workloads = client.list_workload(uuid=workload_uuid).data
+    wl = workloads[0]
+    while wl['state'] != "active":
+        if time.time() - start > timeout:
+            raise AssertionError(
+                "Timed out waiting for state to get to active")
+        time.sleep(.5)
+        workloads = client.list_workload(uuid=workload_uuid).data
+        assert len(workloads) == 1
+        wl = workloads[0]
+    return wl
+
+
+def create_and_verify_isito_app(p_client, ns, project, name):
+    answers = {
+        "enableCRDs": "true",
+        "gateways.istio-ingressgateway.type": "NodePort"
```

Comment:
Is it possible to specify the port? When running tests against our provisioned Azure cluster we will need to specifically open the port in the node template.